packages/web-components/fast-components/src/styles/density.ts

```diff
@@ -0,0 +1,249 @@
+import { composedParent, DesignToken } from "@microsoft/fast-foundation";
+import { designUnit, strokeWidth } from "../design-tokens";
+
+const { create } = DesignToken;
+
+/**
+ * A Design Token representing the visual height of a single line of `base`-sized text.
+ *
+ * @remarks
+ * A variant on the type ramp for the common usage of a single line of text as in most components, adjusted for visual size rather than leading between multiple lines.
+ *
+ * For text-only and icon-only buttons to be the same height, this value should be the same as the icon size.
+ *
+ * @defaultValue 16px
+ *
+ * @public
+ */
+export const typeRampBaseVisualHeight = create<string>(
+    "type-ramp-base-visual-height"
+).withDefault("16px");
+
+// Component horizontal tokens
+
+/**
+ * The design unit multiplier for component density horizontal outer padding.
```

Comment:
Is this representative of space *outside* the component? If so, I have significant concerns around an approach that specifies anything relating to the positioning or spacing between controls. As a best practice, primitives should make no assumptions about their layout or relationship to other elements. That should be applied contextually.

If this isn't related to things outside the component, we should probably work on the semantics. These are quite a mouthful :)